hcl2template/types.packer_config.go

```diff
@@ -119,11 +118,13 @@ func (cfg *PackerConfig) EvalContext(ctx BlockContext, variables map[string]cty.
 		},
 	}
 
-	// Store the iteration_id, if it exists. Otherwise, it'll be "unknown"
-	if cfg.Bucket != nil {
+	iterID, ok := cfg.HCPVars["iterationID"]
+	if ok {
+		log.Printf("iterationID set: %q", iterID)
+
 		ectx.Variables[packerAccessor] = cty.ObjectVal(map[string]cty.Value{
 			"version":     cty.StringVal(cfg.CorePackerVersionString),
-			"iterationID": cty.StringVal(cfg.Bucket.Iteration.ID),
+			"iterationID": iterID,
```

Comment:
We could change this eventually, but this would be a breaking change if we moved it directly.
I guess we should probably first define it in both places, and show a warning stating this is deprecated, and will be removed later on.
I guess having an `hcp` accessor is relevant if we want to give access to more information that comes back from HCP later on, so we could probably do it.
I think it can be done in a later PR however